### PR TITLE
Fix name trim

### PR DESCRIPTION
**Pull request by EugeoSynthesisThirtyTwo**
Sorry for my english, I am french.

Currently, when the cleanUpMessage function finds "{{user}}:" or "{{char}}:" at the beginning of a generation, it simply destroys the whole string instead of just removing the prefix. It caused problems for the image generation extension because most of the replies from the AI would start with "{{char}}:" and the whole string would be destroyed causing the extension to say "The model produced no response, please retry" or something similar.

I fixed that.

Also I choosed to merged it in the branch "staging" but I don't know if it's the good way to do it.

### CODE REVIEW

File: public/script.js [hunk 1]
```diff
@@ -4411,8 +4411,9 @@ function cleanUpMessage(getMessage, isImpersonate, isContinue, displayIncomplete
         nameToTrim = power_user.allow_name1_display ? '' : name1;
     }
 
-    if (nameToTrim && getMessage.indexOf(`${nameToTrim}:`) == 0) {
-        getMessage = getMessage.substring(0, getMessage.indexOf(`${nameToTrim}:`));
+    if (nameToTrim && getMessage.startsWith(nameToTrim + ':')) {
```
Comment (Cohee1207) [inline public/script.js:4414]: nameToTrim in this context will be `{{user}}:`, not `{{char}}:`

**Comment by Cohee1207** [comment]
Try this on the latest release, it should have prompt generation templates adjusted to enforce a neutral name in the prompt, not user or char.

**Comment by EugeoSynthesisThirtyTwo** [comment]
At least with the few tests I have made, I don't have this issue anymore on the latest version. It seems like the image generation module forces the prompt to start with "System:" preventing the AI to write "{{char}}:"

But the `cleanupMessage` function is still the same, and I still don't understand why someone would write
```js
    if (nameToTrim && getMessage.indexOf(`${nameToTrim}:`) == 0) {
        getMessage = getMessage.substring(0, getMessage.indexOf(`${nameToTrim}:`));
    }
```

since it's equivalent to 
```js
    if (nameToTrim && getMessage.startsWith(`${nameToTrim}:`)) {
        getMessage = "";
    }
```

AND since at the end of the function there is that (which makes more sense and seems like the intended behavior for `nameToTrim` as well)
```js
    if (getMessage.startsWith(nameToTrim2 + ':')) {
        getMessage = getMessage.replace(nameToTrim2 + ':', '');
        getMessage = getMessage.trimStart();
    }
```

**Comment by Cohee1207** [comment]
This is fine to replace getMessage = substring(0, 0) with getMessage = "".

This case exists specifically to prevent model impersonation when the model starts answering with the user name prefix, essentially producing a blank reply and forcing ST to regenerate the message (only in non-streaming or background prompt mode). 

But nameToTrim and nameToTrim2 are not equivalent. The second one exists to remove the leading character name from the reply if it wasn't requested by the user to stay there, for example, name2 is Alice and the module produced:
```
Alice: Hi
```
When the user setting "`Allow {{char}} is responses`" is disabled (its default state), the string will be trimmed to:
```
Hi
```
Otherwise, it will be allowed to stay as is.

**Comment by EugeoSynthesisThirtyTwo** [comment]
Ok thanks. How can I change "Allow {{char}} in responses" please ?

**Comment by Cohee1207** [comment]
It's under User settings.

**Pull request closed without merging**